x-pack/plugins/security_solution/public/detections/components/callouts/missing_privileges_callout/translations.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  DEFAULT_ITEMS_INDEX,
+  DEFAULT_LISTS_INDEX,
+  DEFAULT_SIGNALS_INDEX,
+  SAVED_OBJECTS_MANAGEMENT_FEATURE_ID,
+} from '../../../../../common/constants';
+import { CommaSeparatedValues } from './comma_separated_values';
+import { MissingPrivileges } from './use_missing_privileges';
+
+export const MISSING_PRIVILEGES_CALLOUT_TITLE = i18n.translate(
+  'xpack.securitySolution.detectionEngine.missingPrivilegesCallOut.messageTitle',
+  {
+    defaultMessage: 'Insufficient privileges',
+  }
+);
+
+const CANNOT_EDIT_RULES = i18n.translate(
+  'xpack.securitySolution.detectionEngine.missingPrivilegesCallOut.cannotEditRules',
+  {
+    defaultMessage: 'Without that privilege you cannot create or edit detection engine rules.',
+  }
+);
+
+const CANNOT_EDIT_LISTS = i18n.translate(
+  'xpack.securitySolution.detectionEngine.missingPrivilegesCallOut.cannotEditLists',
+  {
+    defaultMessage: 'Without these privileges, you cannot create or edit value lists.',
+  }
+);
+
+const CANNOT_EDIT_ALERTS = i18n.translate(
+  'xpack.securitySolution.detectionEngine.missingPrivilegesCallOut.cannotEditAlerts',
+  {
+    defaultMessage: 'Without these privileges, you cannot open or close alerts.',
+  }
+);
+
+export const missingPrivilegesCallOutBody = ({
+  indexPrivileges,
+  featurePrivileges,
+}: MissingPrivileges) => (
+  <FormattedMessage
+    id="xpack.securitySolution.detectionEngine.missingPrivilegesCallOut.messageBody.messageDetail"
+    defaultMessage="{essence} Missing privileges: {privileges} Related documentation: {docs}"
+    values={{
+      essence: (
+        <p>
+          <FormattedMessage
+            id="xpack.securitySolution.detectionEngine.missingPrivilegesCallOut.messageBody.essenceDescription"
+            defaultMessage="You need the following privileges to fully access this functionality. Contact your administrator for further assistance."
+          />
+        </p>
+      ),
+      privileges: (
+        <ul>
+          {indexPrivileges.map(([index, missingPrivileges]) => (
+            <li key={index}>{missingIndexPrivileges(index, missingPrivileges)}</li>
+          ))}
+          {featurePrivileges.map(([feature, missingPrivileges]) => (
+            <li key={feature}>{missingFeaturePrivileges(feature, missingPrivileges)}</li>
+          ))}
+        </ul>
+      ),
+      docs: (
+        <ul>
+          <li>
+            <DetectionsRequirementsLink />
+          </li>
+          <li>
+            <SecuritySolutionRequirementsLink />
+          </li>
+        </ul>
+      ),
+    }}
+  />
+);
+
+interface PrivilegeExplanations {
+  [key: string]: {
+    [privilegeName: string]: string;
+  };
+}
+
+const PRIVILEGE_EXPLANATIONS: PrivilegeExplanations = {
+  [SAVED_OBJECTS_MANAGEMENT_FEATURE_ID]: {
+    all: CANNOT_EDIT_RULES,
+  },
+  [DEFAULT_SIGNALS_INDEX]: {
+    write: CANNOT_EDIT_ALERTS,
+  },
+  [DEFAULT_LISTS_INDEX]: {
+    write: CANNOT_EDIT_LISTS,
+  },
+  [DEFAULT_ITEMS_INDEX]: {
+    write: CANNOT_EDIT_LISTS,
+  },
```

Comment:
That's a good point. This logic works with custom indices so long as they start with `.siem-signals*`, `.lists*`, `.items*` because I check them with `startsWith`. But with completely custom index names, it wouldn't work. How to know whether we support them or no?

Reply:
For visibility, feedback regarding that:

> they're not officially supported and so left out of the docs. That said the signals index override has been mentioned a few times in the community [1] and the lists settings [2] are called out in the README.md, but they're technically unsupported features
> 
> [1] https://discuss.elastic.co/t/detection-result-in-new-index/270347
> [2] https://github.com/elastic/kibana/blob/master/x-pack/plugins/lists/README.md